docs/pages/access-controls/guides/hardware-key-support.mdx

```diff
@@ -0,0 +1,215 @@
+    - Application Access
+    - Desktop Access
+  
+  Hardware-backed Login sessions do not yet support:
+
+    - Teleport WebUI (except for user registration / reset password)
+    - Agent forwarding functionality such as `tsh ssh -A`, Proxy Recording mode, and OpenSSH integration
+    - Kubernetes Access
+
+  To navigate these incompatiblities, we recommend enabled Hardware Key support only when necessary, such as for roles with access to critical infrastructure. These roles can be accessed as needed with Access Requests so that users can avoid these issues for their normal login sessions.
+
+</Admonition>
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A series 5+ YubiKey
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/2. Enable Hardware Key support
+
+Hardware Key support is disbled by default. You can enable Hardware Key support for specific roles like below:
+
+```yaml
+kind: role
+metadata:
+  name: admin
+spec:
+  allow:
+    require_session_mfa: hardware_key_touch
+```
+
+You can also enabled Hardware Key support cluster-wide by updating yout Teleport configuration as below:
+
+<ScopedBlock scope={["oss", "enterprise"]}>
+<Tabs>
+  <TabItem label="Static Config">
+    Auth Server `teleport.yaml` file:
+
+    ```yaml
+    # snippet from /etc/teleport.yaml:
+    auth_service:
+      authentication:
+        type: local
+        require_session_mfa: hardware_key_touch
+    ```
+  </TabItem>
+  <TabItem label="Dynamic resources">
+
+  Obtain your existing `cluster_auth_preference` resource:
+
+  ```code
+  $ tctl get cap > cap.yaml
+  ```
+
+  If you have not defined a `cluster_auth_preference`, `cap.yaml` will be blank.
+
+  Ensure that `cap.yaml` includes the following content:
+
+  ```yaml
+  kind: cluster_auth_preference
+  version: v2
+  metadata:
+    name: cluster-auth-preference
+  spec:
+    type: local
+    require_session_mfa: hardware_key_touch
+  ```
+
+  Update the configuration:
+
+  ```code
+  $ tctl create -f cap.yaml
+  # cluster auth preference has been updated
+  ```
+
+  </TabItem>
+</Tabs>
+</ScopedBlock>
+
+<ScopedBlock scope={["cloud"]}>
+
+Obtain your existing `cluster_auth_preference` resource:
+
+```code
+$ tctl get cap > cap.yaml
+```
+
+If you have not defined a `cluster_auth_preference`, `cap.yaml` will be blank.
+
+Ensure that `cap.yaml` includes the following content:
+
+```yaml
+kind: cluster_auth_preference
+version: v2
+metadata:
+  name: cluster-auth-preference
+spec:
+  type: local
+  require_session_mfa: hardware_key_touch
+```
+
+Update the configuration:
+
+```code
+$ tctl create -f cap.yaml
+# cluster auth preference has been updated
+```
+
+</ScopedBlock>
+
+## Step 2/2. Log in
+
+Once enabled on the Auth server, affected users will be prompted to connect and touch their YubiKey on login:
+
+<ScopedBlock scope={["oss", "enterprise"]}>
+
+```code
+$ tsh login --user=dev --proxy=proxy.example.com:3080
+# Enter password for Teleport user dev:
+# Tap your YubiKey
```

Comment:
Similar to above comments - did we actually hard-code "YubiKey" in the source code? Should we have?

Reply:
As Brian mentioned above. We want to use YubiKey in user messages, as PIV is not a very well-known standard. Currently, YubiKey is the only popular key that uses it. We also didn't test anything else.